src/tensorflow.cc

```diff
@@ -1230,14 +1224,24 @@ AutoCompleteHelper::FixBatchingSupport()
             .c_str());
   }
 
-  // 'model_support_batching_' is set to be true when all model inputs have
-  // variable size first dimension, but it is not necessary to be the case
-  // (i.e. non-batch model with variable size tensors). As 'max_batch_size == 0'
-  // from existing config is also ambiguous, it can be either unspecified or
-  // no-batch, autofill will check specified input/output (if any) for hint.
-  model_support_batching_ = sig_supports_batch;
-  if (model_support_batching_) {
-    bool config_batch_hint = false;
+  bool config_batch_hint = false;
+  if (sig_supports_batch) {
+    // When the rank of an I/O is greater than 0 and the 
+    // model config I/O dimensions has size greater than 
+    // 0, then this is a strong hint at for/against batching.
+    enum struct ExplicitBatchHint {
+      UNSET,
+      TRUE,
+      FALSE
+    };
+    ExplicitBatchHint config_explicit_hint = ExplicitBatchHint::UNSET;
+
+    // 'model_support_batching_' is set to be true when all model inputs have
+    // variable size first dimension, but it is not necessary to be the case
+    // (i.e. non-batch model with variable size tensors). As 'max_batch_size == 0'
+    // from existing config is also ambiguous, it can be either unspecified or
+    // no-batch, autofill will check specified input/output (if any) for hint.
+    config_batch_hint = false;
```

Comment:
This variable is already initialized as false in L1227. Do you want to move the declaration of `config_batch_hint` from there to here?

Reply:
removed the line as the L1227 declaration is used after the following if-statement